chain/epoch-manager/src/shard_tracker.rs

```diff
@@ -129,8 +74,24 @@ impl ShardTracker {
         is_me: bool,
         epoch_selection: EpochSelection,
     ) -> bool {
-        // TODO: fix these unwrap_or here and handle error correctly. The current behavior masks potential errors and bugs
-        // https://github.com/near/nearcore/issues/4936
+        // Compute epoch_id for all cases
+        let epoch_id = match epoch_selection {
```

Comment:
Let's move it to a separate method like `get_epoch_id` and let `cares_about_shard_in_epoch` look like this:

```
fn cares_about_shard_in_epoch:
  if cfg!(debug_assertions) {
      let epoch_id = self.get_epoch_id(...).expect("...");
      self.check_shard_exists_in_epoch(...);
  }
  ...
  ...
  ...
  let Ok(epoch_id) = self.get_epoch_id(parent_hash, epoch_selection) else {
      return false;
  };
  self.tracks_shard_at_epoch(shard_id, &epoch_id);
}
```